src/frogger.cpp

```diff
@@ -0,0 +1,77 @@
+#include <Rcpp.h>
+
+#include "frogger.hpp"
+#include "types.hpp"
+
+int get_simulation_years(const Rcpp::List demp, SEXP r_sim_years) {
+  Rcpp::NumericVector Sx = demp["Sx"];
+  Rcpp::Dimension d = Sx.attr("dim");
+  // Simulation initialises state from first years input data (index 0)
+  // then runs for each year simulating this years (i) data using previous years
+  // state (i - 1) and this years input data (i). So -1 off index for max years
+  // to simulate as index 0 used for initial state
+  const int max_sim_years = d[2] - 1;
+  if (r_sim_years == R_NilValue) {
+    return max_sim_years;
+  }
+  auto sim_years = INTEGER(r_sim_years)[0];
+  if (sim_years > max_sim_years) {
+    Rcpp::stop("No of years > max years of " + std::to_string(max_sim_years));
+  }
+  return sim_years;
+}
+
+// [[Rcpp::export]]
+Rcpp::List run_base_model(const Rcpp::List demp, SEXP sim_years) {
+  const int proj_years = get_simulation_years(demp, sim_years);
+  const int num_genders = 2;
+  const int age_groups_pop = 81;
+  const int fertility_first_age_group = 15;
+  const int age_groups_fert = 35;
+
+  TensorMap2<double> base_pop(REAL(demp["basepop"]), age_groups_pop,
```

Comment:
At some point we will need validation of the sizes here or we'll risk crashes

Reply:
Thanks, created a ticket https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-3301